lib/puppet/interface.rb

```diff
@@ -154,10 +154,7 @@ def initialize(name, version, &block)
     end
 
     @name    = Puppet::Interface::FaceCollection.underscorize(name)
-
-    # SemVer is deprecated but in 4.x we must use it here (the attr_reader is public api). The
-    # extra boolean argument suppresses the deprecation warning.
-    @version = SemVer.new(version, true)
+    @version = SemanticPuppet::Version.parse(version)
```

Comment:
@Iristyle and I talked about this, and this doesn't appear to hurt anything. It switches PMT to using semantic puppet that is already vendored in puppet (so there's no integration issue for puppetserver) and there's a perf benefit, so 👍 to including in 4.10.x